umbraco/Umbraco-CMS
# V8/bugfix/3498 entity service updates

Fixes #3498 

Please read notes in the tasks and related links.

The gist is: 

* The EntityService for media was querying all media properties which is slow and shouldn't be done (here's a description of why it was loading in all props: https://github.com/umbraco/Umbraco-CMS/pull/3460#issuecomment-434903930)
* It did this in order to get the media's file path, but we can do this a nicer way by joining the `umbracoMediaVersion` table which contains the media path

So this is all done and tons of code is cleaned up/removed. I'd wager that nobody was ever using the code that was removed. There would be no reason why anyone was using these services and passing in a true value for `loadBaseType` (or `full`). 

The `_mediaFileResolvers` stuff that was mentioned in one of those links still remains and that is fine, they are still used to resolve a media's path based on a full media item (i.e. when you click on a media folder in the media section... that said ... it would be much much faster and better performance to load in all media items and thumbnails when you click on a folder to use the EntityService, something to consider in the future since now it uses the MediaService which is loading in all property data!)

### Testing

* Make sure that the media picker still works and shows a thumbnail
* Make sure that the media grid works and shows thumbnails when you click on a folder in the media tree that contains images
* Make sure unit tests are all passing
* review code
